web/search.dart

```diff
@@ -203,24 +210,53 @@ void initializeSearch(
         query, "<strong class='tt-highlight'>$sanitizedText</strong>");
   }
 
-  Element createSuggestion(String query, IndexItem match) {
-    var suggestion = document.createElement('div');
-    suggestion.setAttribute('data-href', match.href ?? '');
-    suggestion.classes.add('tt-suggestion');
+  var categoriesMap = <String, Element>{};
 
-    var suggestionTitle = document.createElement('span');
-    suggestionTitle.classes.add('tt-suggestion-title');
-    suggestionTitle.innerHtml =
-        highlight('${match.name} ${match.type.toLowerCase()}', query);
+  // Maps a suggestions library/class to the other suggestions that search found if any.
+  void categorize(Element suggestionLibrary, Element suggestion) {
+    var input = suggestionLibrary.innerHtml;
 
-    suggestion.append(suggestionTitle);
+    if (input == null) {
+      return;
+    }
 
-    if (match.enclosedBy != null) {
-      var fromLib = document.createElement('div');
-      fromLib.classes.add('search-from-lib');
-      fromLib.innerHtml = 'from ${highlight(match.enclosedBy!.name, query)}';
+    var element = categoriesMap[input];
+    if (element != null) {
+      element.append(suggestion);
+    } else {
+      suggestionLibrary.append(suggestion);
+      categoriesMap[input] = suggestionLibrary;
+    }
+  }
 
-      suggestion.append(fromLib);
+  // Using the name of the library/class creates the Element for it.
+  Element createCategory(String lib, String href) {
```

Comment:
Instead of "Category", which is a different Dartdoc concept, could we use a similar word, like Group or Grouping or something similar.